bootstrap/kubeadm/config/default/kustomizeconfig.yaml

```diff
@@ -0,0 +1,4 @@
+# This configuration is for teaching kustomize how to update name ref and var substitution
+varReference:
+- kind: Deployment
+  path: spec/template/spec/volumes/secret/secretName
```

Comment:
Is this actually needed? I feel like this was something that we previously used to hack together the separate manager and webhook deployments. I'm  not seeing this in the upstream kubuebuilder examples: https://github.com/kubernetes-sigs/kubebuilder/tree/63b35350fe8b4d133c0bdb8fdd9a9a61fe30ac61/testdata/project-v3/config/default

Reply:
This is not in the kubuilder example, but I felt that we are on the safe path if we ensure all the webhook services get a different name for each provider, so during the reshuffle I preserved:

```
secretName: $(SERVICE_NAME)-cert # this secret will not be prefixed since it's not managed by kustomize
```

in `bootstrap/kubeadm/config/certmanager/certificate.yaml` and made the necessary changes for it being resolved in `config/default/kustomize.yaml` which basically consist of creating this file and referencing it